docs/metrics-data-model.md

```diff
@@ -0,0 +1,177 @@
+# k6 Time Series
+
+We would migrate to a time series data model for getting the benefits of aggregated data so just allocating samples one time and pass around the aggregated time series value (or values - e.g. in case of Trend). The Outputs will switch from fetching the buffered samples to fetching a snapshot of the time series (the same concept adopted by Prometheus' scraping operation).
+
+## Data model
+
+#### Metrics and Series
+
+* Metric (name, type)
+* TimeSeries (metric, tagset)
+* Sample (metric, tagset, timestamp, value)
+* TimeSeries Sample (timeseries ref, timestamp, value)
+
+```go
+// This is the current k6 Metric polished from the Thresholds stuff.
+type Metric struct {
+	Name     string     `json:"name"`
+	Type     MetricType `json:"type"`
+	Contains ValueType  `json:"contains"`
```

Comment:
```suggestion
* Do we need `Contains ValueType`?
```
It seems used only from the summary. Should we move the mapping directly there? If not, then I think we should extend the [current JS API](https://k6.io/docs/javascript-api/k6-metrics/trend/) for supporting all the possible values (currently, it supports only Time).